src/fmc.rs

```diff
@@ -0,0 +1,278 @@
+//! HAL for Flexible memory controller (FMC)
+
+// From stm32_fmc
+use stm32_fmc::FmcPeripheral;
+use stm32_fmc::{PinsSdram, Sdram, SdramChip, SdramPinSet, SdramTargetBank};
+
+use crate::rcc::{rec, rec::ResetEnable, CoreClocks};
+use crate::stm32;
+use crate::time::Hertz;
+
+use crate::gpio::gpioa::PA7;
+use crate::gpio::gpiob::{PB5, PB6, PB7};
+use crate::gpio::gpioc::{PC0, PC2, PC3, PC4, PC5, PC6, PC7, PC8};
+use crate::gpio::gpiod::{
+    PD0, PD1, PD10, PD11, PD12, PD13, PD14, PD15, PD3, PD4, PD5, PD6, PD7, PD8,
+    PD9,
+};
+use crate::gpio::gpioe::{
+    PE0, PE1, PE10, PE11, PE12, PE13, PE14, PE15, PE2, PE3, PE4, PE5, PE6, PE7,
+    PE8, PE9,
+};
+use crate::gpio::gpiof::{
+    PF0, PF1, PF11, PF12, PF13, PF14, PF15, PF2, PF3, PF4, PF5,
+};
+use crate::gpio::gpiog::{
+    PG0, PG1, PG10, PG12, PG13, PG14, PG15, PG2, PG3, PG4, PG5, PG6, PG7, PG8,
+    PG9,
+};
+use crate::gpio::gpioh::{
+    PH10, PH11, PH12, PH13, PH14, PH15, PH2, PH3, PH5, PH6, PH7, PH8, PH9,
+};
+use crate::gpio::gpioi::{PI0, PI1, PI10, PI2, PI3, PI4, PI5, PI6, PI7, PI9};
+use crate::gpio::{Alternate, AF12, AF9};
+
+/// Storage type for Flexible Memory Controller and its clocks
+pub struct FMC {
+    fmc: stm32::FMC,
+    _prec: rec::Fmc,
```

Comment:
Why even bother storing the PREC if you're not going to use it? Alternatively, I think you should be able to use it in the `enable()` function instead of taking direct access to the RCC.